### DISCUSSION TITLE
CentOS 7 kernel security update - Roadmap for integrating into Security Onion?

### DISCUSSION THREAD

**Discussion by ghost** [Security-Onion-Solutions/securityonion#2491] [Unsupported Versions] [answered]
Hello. My local Qualys install notified me of a CentOS 7 vulnerability:

https://lists.centos.org/pipermail/centos-announce/2020-December/048241.html

Description of update is here: https://access.redhat.com/errata/RHSA-2020:5437.

The CVSS scores in the moderate range; the highest I saw was 7.1. The rest are in the 5's to 6's. Not "Critical", thank goodness, but also not anything to ignore.

That said, this is being identified by Qualys as a "Severity 4" (scale of 1 to 5) vulnerability. So while this doesn't affect me personally in my lab/test environment, anyone else with a production Security Onion service that's based on the iso - and who's also running Qualys or anything else that may identify this - may need to take some action to satisfy their internal audit or compliance staff. 

So bottom line: Is there any time frame for updating the iso's OS with these updates?

Thank you.

**Comment by dougburks** [comment]
Which ISO image did you install?

When the installer asked you about patch schedule, what did you select?
![image](https://user-images.githubusercontent.com/1659467/103316831-af326f80-49f7-11eb-9b40-97c963948be3.png)

Have you tried running `sudo yum update`?

**Comment by ghost** [comment]
I installed 2.3.10 and have since updated it to current (2.3.21). I also selected Automatic upon install.

Soup was telling me there are no updates, and neither did Yum (yes, I did try that earlier), but upon seeing your question, I dug a bit deeper. 

![Screenshot mgrnode](https://user-images.githubusercontent.com/60711445/103320774-b57b1880-4a04-11eb-841a-f82c1d355b0f.png)

Aha. Both the vulnerable old kernel listed in the scan report _and_ the newest required versions are installed. I'm thinking that has to be the issue. I'll uninstall the old kernel and find out how to make Qualys agent rescan. I'll update when that's complete, but I really suspect now that this is a Qualys detection issue and not a Soup or Yum one. 

Yeah, I should've run that down before I posted. My mistake. I'll confirm here soon.

**Comment by dougburks** [comment]
Perhaps the box hasn't been rebooted since the new kernel was installed and Qualys is looking at the running kernel?

**Reply by ghost** [reply]
Maybe it did need a reboot, I don't know, but the old kernel shouldn't have been the running one. Uname listed only the newest one. 

I now suspect that instead of uname, Qualys does either the rpm -qa kernel or yum list installed kernel to see what's installed. I didn't think it would go that far, but it clearly picked up on the presence of the not-running kernel.

I've initiated a new Qualys agent scan. I'll see what comes of this.

**Comment by ghost** [comment, marked as answer]
Ok, confirmed, the report in Qualys is gone now that the old kernel was removed. And Security Onion's CentOS install was indeed updating properly. We can close this thread. I should've checked before presuming there was either a version lock or something weird with Soup. That's my fault.
